src/rust/src/transaction_ffi.rs

```diff
@@ -261,6 +262,21 @@ pub extern "C" fn zcash_transaction_zip244_signature_digest(
     let signable_input = if index == NOT_AN_INPUT {
         SignableInput::Shielded
     } else {
+        // This conversion to `u8` is always fine:
+        // - We only call this FFI method once we already know we are using ZIP 244.
+        // - Even if we weren't, `hash_type` is one byte tacked onto the end of a
+        //   signature, so it always fits into a `u8` (and TBH I don't know why we
+        //   ever set it to `u32`).
```

Comment:
That's correct (the only caller first checks that `nHashType` is valid and therefore fits in `u8`), but please document the panic. While you're at it, might as well use the appropriate doc convention for safety requirements.